### DISCUSSION TITLE
Mlflow.evaluate:Cannot interpret 'CategoricalDtype(categories=[0, 1], ordered=False)' as a data type

### DISCUSSION THREAD

**Discussion by khojarohan4** [mlflow/mlflow#6346] [Q&A]
Trying to use Mlflow evaluate for Histogram based gradient boosting classifier but keep getting subjected error. Not sure what is wrong.

My data is like below:

X_test has 27 features of which 25 are float64 and other 2 are of type category
Y_test is used as label with type as int64

I have added y_test inside my eval_data and tried to run mlflow evaluate but I keep getting the subjected error. Cant my X_test have dtype of category for this evaluate method to run?

`File ~/opt/anaconda3/lib/python3.9/site-packages/mlflow/models/evaluation/base.py:636, in _evaluate(model, model_type, dataset, run_id, evaluator_name_list, evaluator_name_to_conf_map, custom_metrics)
    634     if evaluator.can_evaluate(model_type=model_type, evaluator_config=config):
    635         _logger.info(f"Evaluating the model with the {evaluator_name} evaluator.")
--> 636         result = evaluator.evaluate(
    637             model=model,
    638             model_type=model_type,
    639             dataset=dataset,
    640             run_id=run_id,
    641             evaluator_config=config,
    642             custom_metrics=custom_metrics,
    643         )
    644         eval_results.append(result)
    646 _last_failed_evaluator = None

File ~/opt/anaconda3/lib/python3.9/site-packages/mlflow/models/evaluation/default_evaluator.py:1032, in DefaultEvaluator.evaluate(self, model, model_type, dataset, run_id, evaluator_config, custom_metrics, **kwargs)
   1030 with mlflow.utils.autologging_utils.disable_autologging():
   1031     if model_type == "classifier":
-> 1032         return self._evaluate_classifier()
   1033     elif model_type == "regressor":
   1034         return self._evaluate_regressor()

File ~/opt/anaconda3/lib/python3.9/site-packages/mlflow/models/evaluation/default_evaluator.py:974, in DefaultEvaluator._evaluate_classifier(self)
    968 if hasattr(sk_metrics, "ConfusionMatrixDisplay"):
    969     self._log_image_artifact(
    970         plot_confusion_matrix,
    971         "confusion_matrix",
    972     )
--> 974 return self._log_and_return_evaluation_result()

File ~/opt/anaconda3/lib/python3.9/site-packages/mlflow/models/evaluation/default_evaluator.py:886, in DefaultEvaluator._log_and_return_evaluation_result(self)
    884 self._evaluate_custom_metrics_and_log_produced_artifacts()
    885 self._log_metrics()
--> 886 self._log_model_explainability()
    887 return EvaluationResult(self.metrics, self.artifacts)

File ~/opt/anaconda3/lib/python3.9/site-packages/mlflow/models/evaluation/default_evaluator.py:512, in DefaultEvaluator._log_model_explainability(self)
    510 feature_dtypes = list(self.X.get_original().dtypes)
    511 for feature_dtype in feature_dtypes:
--> 512     if not np.issubdtype(feature_dtype, np.number):
    513         _logger.warning(
    514             "Skip logging model explainability insights because the shap explainer "
    515             f"{algorithm} requires all feature values to be numeric, and each feature "
    516             "column must only contain scalar values."
    517         )
    518         return

File ~/opt/anaconda3/lib/python3.9/site-packages/numpy/core/numerictypes.py:418, in issubdtype(arg1, arg2)
    360 r"""
    361 Returns True if first argument is a typecode lower/equal in type hierarchy.
    362 
   (...)
    415 
    416 """
    417 if not issubclass_(arg1, generic):
--> 418     arg1 = dtype(arg1).type
    419 if not issubclass_(arg2, generic):
    420     arg2 = dtype(arg2).type

TypeError: Cannot interpret 'CategoricalDtype(categories=[0, 1], ordered=False)' as a data type`

**Comment by khojarohan4** [comment]
I was able to bypass the error by providing paramters to evaluate as below but curious to know what causes this and the actual way around it?

evaluators=["default"],
evaluator_config={"default":{"log_model_explainability":False}}

**Reply by WeichenXu123** [reply]
The default model shap explainability does not support CategoricalDtype.
I remember the latest mlflow version has add try .. except to capture the exception in shap explainability evaluation.

**Reply by khojarohan4** [reply]
Ok so I can't get shap related visualization until I have features with categorical dtype? Is this understanding accurate?

**Reply by WeichenXu123** [reply]
For the default shap explainer, To get the shap visualization plot, you need to avoid use categorical dtype features.
